### PR TITLE
SL-219 Saving cards

### PR DESCRIPTION
**Pull request by MarijusCoding**
Fixes:
* Removed useless redirection inside iframe
* Added check is customer trying to save same card
* Added SQL for gathering only `card_number` from card alias

### CODE REVIEW

File: src/Api/Request/AuthorizationService.php [hunk 1]
```diff
@@ -103,6 +108,13 @@ public function createObjectsFromAuthorizationResponse(
         $assertBody = $this->assertResponseObjectCreator->createAssertObject($responseBody);
         $this->assertBuilder->createAssert($assertBody, $saferPayOrderId);
         $isPaymentSafe = $assertBody->getLiability()->getLiabilityShift();
+        $allSavedCards = array_column($this->aliasRepository->getSavedCardsNumbersByCustomerId($customerId), 'card_number');
+        $currentCard = $assertBody->getPaymentMeans()->getDisplayText();
+
+        if (in_array($currentCard, $allSavedCards)) {
```
Comment (joelhelvig) [inline src/Api/Request/AuthorizationService.php:114]: Hello, is this check preventing hackers to use someone else saved card ?

Comment (MarijusCoding) [reply src/Api/Request/AuthorizationService.php:114]: Hello @joelhelvig
Thank you for asking!

This logic provides check if is current user card is saved for user. To not have same cards saved in Account page.
Answering your question, we don't have any sensitive card information, we only have last 4 numbers of card `xxxx xxxx xxxx 1234` and that's all we got from customer inputs. No sensitive data is being manipulated threw Back-end.

If you have any more questions, let us know.

Have a good day!

File: src/Repository/SaferPayCardAliasRepository.php [hunk 2]
```diff
@@ -74,4 +74,14 @@ public function getSavedCardsByCustomerId($customerId)
...
     }
+
+    public function getSavedCardsNumbersByCustomerId($customerId)
+    {
+        $query = new DbQuery();
+        $query->select('`card_number`');
+        $query->from('saferpay_card_alias');
+        $query->where('id_customer = "' . (int) $customerId . '"');
+
+        return Db::getInstance()->executeS($query);
```
Comment (GytisZum) [inline src/Repository/SaferPayCardAliasRepository.php:85]: What if result is null ? Make sure have consistent return types, would be nice to have phpDocs above methods

You should check is result legit, if no return empty array

File: src/Repository/SaferPayCardAliasRepository.php [hunk 3]
```diff
@@ -41,9 +48,13 @@ public function getSavedValidCardsByUserIdAndPaymentMethod($userId, $paymentMeth
         $query->where('payment_method = "' . pSQL($paymentMethod) . '"');
         $query->where('valid_till > "' . pSQL($currentDate) . '"');
 
-        return Db::getInstance()->executeS($query);
+        return (array) Db::getInstance()->executeS($query) ?? [];
```
Comment (GytisZum) [inline src/Repository/SaferPayCardAliasRepository.php:51]: null coalescing operator was presented in PHP 7, module compatible with 5.6

**Review by GytisZum — CHANGES_REQUESTED** [review]

**Comment by MarijusCoding** [comment]
Broken PR :'(

**Pull request closed without merging**